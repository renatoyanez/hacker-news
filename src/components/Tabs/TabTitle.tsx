import { useCallback } from 'react';
import styles from './tabs.module.css';

export type TabTitleProps = {
	title: string;
	index: number;
	setSelectedTab: (index: number) => void;
	isActive?: boolean;
};

const TabTitle = (props: TabTitleProps): JSX.Element => {
	const { title, setSelectedTab, index, isActive } = props;

	const handleOnClick = useCallback(() => {
		setSelectedTab(index);
	}, [setSelectedTab, index]);

	return (
		<li className={`${styles.title} ${isActive ? 'active' : ''}`}>
			<button onClick={handleOnClick}>{title}</button>
		</li>
	);
};

export default TabTitle;
