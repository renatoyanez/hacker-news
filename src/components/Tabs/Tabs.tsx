import React from 'react';
import { useNavigate } from 'react-router';
import { useLocation } from 'react-router-dom';
import { Typography } from 'components';
import styles from './tabs.module.css';

interface Elements {
	title: string;
	path: string;
}

interface TabsProps {
	data: Elements[];
}

const Tabs = ({ data }: TabsProps) => {
	const navigate = useNavigate();
	const location = useLocation();
	const pathname = location.pathname;

	const isSelected = (currentPath: string, selectedPath: string) => {
		return currentPath === selectedPath;
	};
	const handleSelect = (path: string) => {
		navigate(path);
	};

	return (
		<div>
			{data.map((item, index) => {
				return (
					<button
						key={item.title}
						onClick={() => handleSelect(item.path)}
						className={`${
							isSelected(pathname, item.path)
								? styles.selectedButton
								: styles.button
						} ${index === 0 ? styles.first : styles.second}`}
					>
						<Typography
							type={
								isSelected(pathname, item.path)
									? 'textStyle2'
									: 'textStyle5'
							}
						>
							{item.title}
						</Typography>
					</button>
				);
			})}
		</div>
	);
};

export default Tabs;
