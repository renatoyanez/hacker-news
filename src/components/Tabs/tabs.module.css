@import '../../styles/variables.css';

.selectedButton {
	width: 6.125rem;
	background-color: transparent;
	border: solid 1px var(--azure);
}

.button {
	width: 6.125rem;
	background-color: transparent;
	border: solid 1px #d6d6d6;
}

.first {
	border-radius: 2px 0px 0px 2px;
}
.second {
	border-radius: 0px 2px 2px 0px;
}
