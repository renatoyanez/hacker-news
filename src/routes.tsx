import { ReactElement } from 'react';
import { Route, Routes as Switch } from 'react-router-dom';
import { Faves, Home } from 'pages';

type RouteListType = {
	HOME: string;
	FAVES: string;
};

export const RouteList: RouteListType = {
	HOME: '/',
	FAVES: '/my-faves',
};

const Routes = (): ReactElement => {
	return (
		<Switch>
			<Route path={RouteList.HOME} element={<Home />} />
			<Route path={RouteList.FAVES} element={<Faves />} />
		</Switch>
	);
};

export default Routes;
