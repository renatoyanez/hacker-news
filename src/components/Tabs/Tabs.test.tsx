import React from 'react';
import { Router as MemRouter } from 'react-router-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import { create } from 'react-test-renderer';
import { createMemoryHistory } from 'history';
import { RouteList } from 'routes';
import Tabs, { TabsProps, Elements } from './Tabs';

const elements: Elements[] = [
	{ title: 'All', path: RouteList.HOME },
	{ title: 'My faves', path: RouteList.FAVES },
];

const props: TabsProps = {
	data: elements,
};

describe('Tabs', () => {
	const history = createMemoryHistory();

	it('renders correctly', () => {
		const tree = create(
			<MemRouter location={history.location} navigator={history}>
				<Tabs {...props} />
			</MemRouter>,
		).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('navigates when clicked', () => {
		render(
			<MemRouter location={history.location} navigator={history}>
				<Tabs {...props} />
			</MemRouter>,
		);
		fireEvent.click(screen.getByText('My faves'));
		expect(history.location.pathname).toBe(RouteList.FAVES);
	});
});
