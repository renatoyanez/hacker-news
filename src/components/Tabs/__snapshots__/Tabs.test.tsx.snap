// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs renders correctly 1`] = `
<div>
  <button
    className="selectedButton first"
    onClick={[Function]}
  >
    <p
      className="textStyle2"
    >
      All
    </p>
  </button>
  <button
    className="button second"
    onClick={[Function]}
  >
    <p
      className="textStyle5"
    >
      My faves
    </p>
  </button>
</div>
`;
